target/linux/ath79/dts/ar9344_netgear_pgzng1.dts

```diff
@@ -0,0 +1,341 @@
+
+	aliases {
+		led-boot = &led_power_green;
+		led-failsafe = &led_power_red;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_green;
+		label-mac-device = &eth0;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	i2c {
+		compatible = "i2c-gpio";
+		sda-gpios = <&gpio 14 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&gpio 13 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		expander0: pca9551@60 {
+			compatible = "nxp,pca9551";
+			reg = <0x60>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			led_power_green: led@0 {
+				function = LED_FUNCTION_BOOT;
+				color = <LED_COLOR_ID_GREEN>;
+				reg = <0>;
+				default-state = "off";
+				type = <PCA955X_TYPE_LED>;
+				chan-name = "green:boot";
+			};
+
+			led_power_red: led@1 {
+				function = LED_FUNCTION_BOOT;
+				color = <LED_COLOR_ID_RED>;
+				reg = <1>;
+				default-state = "off";
+				type = <PCA955X_TYPE_LED>;
+				chan-name = "red:boot";
+			};
+
+			led@2 {
+				reg = <2>;
+				default-state = "off";
+				color = <LED_COLOR_ID_GREEN>;
+				function = LED_FUNCTION_WAN;
+				type = <PCA955X_TYPE_LED>;
+			};
+
+			led@3 {
+				reg = <3>;
+				default-state = "off";
+				color = <LED_COLOR_ID_RED>;
+				function = LED_FUNCTION_WAN;
+				type = <PCA955X_TYPE_LED>;
+			};
+
+			led@4 {
+				reg = <4>;
+				default-state = "off";
+				color = <LED_COLOR_ID_GREEN>;
+				function = LED_FUNCTION_WLAN;
+				type = <PCA955X_TYPE_LED>;
+			};
+
+			led@5 {
+				reg = <5>;
+				default-state = "off";
+				color = <LED_COLOR_ID_RED>;
+				function = LED_FUNCTION_WLAN;
+				type = <PCA955X_TYPE_LED>;
+			};
+
+			led@6 {
+				reg = <6>;
+				default-state = "off";
+				color = <LED_COLOR_ID_GREEN>;
+				function = LED_FUNCTION_INDICATOR;
+				type = <PCA955X_TYPE_LED>;
+			};
+
+			led@7 {
+				reg = <7>;
+				default-state = "off";
+				color = <LED_COLOR_ID_RED>;
+				function = LED_FUNCTION_INDICATOR;
+				type = <PCA955X_TYPE_LED>;
+			};
+		};
+
+		expander1: pca9551@61 { 
+			compatible = "nxp,pca9551";
+			reg = <0x61>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			gpio-controller;
+			#gpio-cells = <2>;
+
+			/* zwave_rst - Resets ZWave */
+			gpio@4 {
+				reg = <4>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+
+			/* em_rst - Unknown */
+			gpio@5 {
+				reg = <5>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+
+			/* tp34 - Test point on PCB? */
+			gpio@6 {
+				reg = <6>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+
+			/* sw_rst - resets SoC */
+			gpio@7 {
+				reg = <7>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+		};
+
+		rtc@6f {
+			compatible = "isil,isl1208";
+			reg = <0x6f>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		eth1_link {
+			label = "green:eth1";
```

Comment:
please add color + function properties just like the leds above :)

you can use the function-enumerator property to differentiate between either ethernet port.

Reply:
Ill try with LED_FUNCTION_WAN/LED_FUNCTION_LAN first since it's just 2 ports that are labeled.